src/components/MatchCriterion/edit/MatchCriterion.test.js

```diff
@@ -0,0 +1,150 @@
+  existingRecordType: 'MARC_HOLDINGS',
+  staticValueType: 'TEXT',
+  incomingRecordLabel: 'testIncomingRecordLabel',
+  existingRecordLabel: 'testExistingRecordLabel',
+  existingRecordFields: [{
+    value: 'testValue_1',
+    label: 'testLabel_1',
+  }],
+  onStaticValueTypeChange: onEachChangeMock,
+  onQualifierSectionChange: onEachChangeMock,
+  changeFormState: onEachChangeMock,
+  formValues: { profile: { matchDetails: [{ existingMatchExpression: { fields: [{ value: 'test value' }] } }] } },
+};
+const matchCriterionWithoutQualifierType = {
+  repeatableIndex: 0,
+  matchDetails: {
+    existingMatchExpression: {
+      dataValueType: 'testDataValueType_1',
+      fields: {
+        label: 'testLabel_1',
+        value: 'testValue_1',
+      },
+      qualifier: {
+        comparisonPart: 'NUMERICS_ONLY',
+        qualifierType: '',
+        qualifierValue: 'testQualifierValue_1',
+      },
+    },
+    incomingMatchExpression: {
+      dataValueType: 'testDataValueType_2',
+      fields: {
+        label: 'testLabel_2',
+        value: 'testValue_2',
+      },
+      qualifier: {
+        comparisonPart: 'NUMERICS_ONLY',
+        qualifierType: '',
+        qualifierValue: 'testQualifierValue_2',
+      },
+    },
+    matchCriterion: 'EXACTLY_MATCHES',
+    existingRecordType: 'INSTANCE',
+    incomingRecordType: 'HOLDINGS',
+  },
+  incomingRecordType: 'MARC_BIBLIOGRAPHIC',
+  existingRecordType: 'MARC_HOLDINGS',
+  staticValueType: 'TEXT',
+  incomingRecordLabel: 'testIncomingRecordLabel',
+  existingRecordLabel: 'testExistingRecordLabel',
+  existingRecordFields: [{
+    value: 'testValue_1',
+    label: 'testLabel_1',
+  }],
+  onStaticValueTypeChange: onEachChangeMock,
+  onQualifierSectionChange: onEachChangeMock,
+  changeFormState: onEachChangeMock,
+  formValues: { profile: { matchDetails: [{ existingMatchExpression: { fields: [{ value: 'test value' }] } }] } },
+};
+
+const renderMatchCriterion = ({
+  repeatableIndex,
+  matchDetails,
+  incomingRecordType,
+  existingRecordType,
+  staticValueType,
+  incomingRecordLabel,
+  existingRecordLabel,
+  existingRecordFields,
+  onStaticValueTypeChange,
+  onQualifierSectionChange,
+  changeFormState,
+  formValues,
+}) => {
+  const component = () => (
+    <MatchCriterion
+      repeatableIndex={repeatableIndex}
+      matchDetails={matchDetails}
+      incomingRecordType={incomingRecordType}
+      existingRecordType={existingRecordType}
+      staticValueType={staticValueType}
+      incomingRecordLabel={incomingRecordLabel}
+      existingRecordLabel={existingRecordLabel}
+      existingRecordFields={existingRecordFields}
+      onStaticValueTypeChange={onStaticValueTypeChange}
+      onQualifierSectionChange={onQualifierSectionChange}
+      changeFormState={changeFormState}
+      formValues={formValues}
+    />
+  );
+
+  return renderWithIntl(renderWithFinalForm(component), translationsProperties);
+};
+
+describe('MatchCriterion edit', () => {
```

Comment:
Mock all sections and check that it re-renders when `existing`/`incoming` value changes and all sections are displayed